cuebot/src/main/java/com/imageworks/spcue/servant/ManageJob.java

```diff
@@ -858,10 +966,16 @@ public void addRenderPartition(JobAddRenderPartRequest request, StreamObserver<J
     public void runFilters(JobRunFiltersRequest request, StreamObserver<JobRunFiltersResponse> responseObserver) {
         try {
             setupJobData(request.getJob());
-            JobDetail jobDetail = jobManager.getJobDetail(job.getJobId());
-            filterManager.runFiltersOnJob(jobDetail);
-            responseObserver.onNext(JobRunFiltersResponse.newBuilder().build());
-            responseObserver.onCompleted();
+            if (isJobFinished()) {
+                responseObserver.onError(Status.FAILED_PRECONDITION
+                        .withDescription("Finished jobs are readonly")
+                        .asRuntimeException());
+            } else {
+                JobDetail jobDetail = jobManager.getJobDetail(job.getJobId());
+                filterManager.runFiltersOnJob(jobDetail);
+                responseObserver.onNext(JobRunFiltersResponse.newBuilder().build());
+                responseObserver.onCompleted();
+            }
```

Comment:
Something like:

```java
            if (attemptChange(responseObserver)) {
                JobDetail jobDetail = jobManager.getJobDetail(job.getJobId());
                filterManager.runFiltersOnJob(jobDetail);
                responseObserver.onNext(JobRunFiltersResponse.newBuilder().build());
                responseObserver.onCompleted();
            }
```

Instead of `isJobFinished` we would have a function that will update the response:

```
private boolean attemptChange(responseObserver ResponseObserver) {
        if (env.getProperty("layer.finished_jobs_readonly", String.class) != null &&
                Objects.equals(env.getProperty("layer.finished_jobs_readonly", String.class), "true")) {
            JobDetail jobDetail = this.jobManager.getJobDetail(this.layer.getJobId());
            if (jobDetail.state == JobState.FINISHED) {
                 responseObserver.onError(Status.FAILED_PRECONDITION
                        .withDescription("Finished jobs are readonly")
                        .asRuntimeException());
                 return false
            }
        }
        return true;
    }
```

Reply:
Added in the most recent commit. Should be ready for the final review